test-suite/src/test/scala/org/scalajs/testsuite/jsinterop/ScalaJSDefinedTest.scala

```diff
@@ -0,0 +1,978 @@
+        @JSName("y")
+        def ySetter_=(v: String): Unit = myY = v + " set"
+      }
+
+      val foo = new PropertiesWithExplicitName
+      expect(foo.answerScala).toEqual(42)
+      expect(foo.xScala).toEqual(3)
+      expect(foo.doubleXScala).toEqual(6)
+      foo.xScala = 23
+      expect(foo.xScala).toEqual(23)
+      expect(foo.doubleXScala).toEqual(46)
+      expect(foo.yGetter).toEqual("hello get")
+      foo.ySetter_=("world")
+      expect(foo.yGetter).toEqual("world set get")
+
+      val dyn = (new PropertiesWithExplicitName).asInstanceOf[js.Dynamic]
+      expect(dyn.answerScala).toBeUndefined
+      expect(js.typeOf(dyn.answer)).toBe("number")
+      expect(dyn.answer).toEqual(42)
+      expect(dyn.x).toEqual(3)
+      expect(dyn.doubleX).toEqual(6)
+      dyn.x = 23
+      expect(dyn.x).toEqual(23)
+      expect(dyn.doubleX).toEqual(46)
+      expect(dyn.y).toEqual("hello get")
+      dyn.y = "world"
+      expect(dyn.y).toEqual("world set get")
+    }
+
+    it("protected properties") {
+      @ScalaJSDefined
+      class ProtectedProperties extends js.Object {
+        protected val x: Int = 42
+        protected[testsuite] val y: Int = 43
+      }
+
+      val foo = new ProtectedProperties
+      expect(foo.y).toEqual(43)
+
+      val dyn = foo.asInstanceOf[js.Dynamic]
+      expect(dyn.x).toEqual(42)
+      expect(dyn.y).toEqual(43)
+    }
+
+    it("simple overloaded methods") {
+      @ScalaJSDefined
+      class SimpleOverloadedMethods extends js.Object {
+        def foo(): Int = 42
+        def foo(x: Int): Int = x*2
+      }
+
+      val foo = new SimpleOverloadedMethods
+      expect(foo.foo()).toEqual(42)
+      expect(foo.foo(3)).toEqual(6)
+
+      val dyn = foo.asInstanceOf[js.Dynamic]
+      expect(js.typeOf(dyn.foo)).toBe("function")
+      expect(dyn.foo()).toEqual(42)
+      expect(dyn.foo(3)).toEqual(6)
+    }
+
+    it("renamed overloaded methods") {
+      @ScalaJSDefined
+      class RenamedOverloadedMethods extends js.Object {
+        @JSName("foobar")
+        def foo(): Int = 42
+        @JSName("foobar")
+        def bar(x: Int): Int = x*2
+      }
+
+      val foo = new RenamedOverloadedMethods
+      expect(foo.foo()).toEqual(42)
+      expect(foo.bar(3)).toEqual(6)
+
+      val dyn = foo.asInstanceOf[js.Dynamic]
+      expect(js.typeOf(dyn.foobar)).toBe("function")
+      expect(dyn.foobar()).toEqual(42)
+      expect(dyn.foobar(3)).toEqual(6)
+    }
+
+    it("default parameters") {
+      @ScalaJSDefined
+      class DefaultParameters extends js.Object {
+        def bar(x: Int, y: Int = 1): Int = x + y
+
+        def foobar(x: Int): Int = bar(x)
+      }
+
+      val foo = new DefaultParameters
+      expect(foo.bar(4, 5)).toEqual(9)
+      expect(foo.bar(4)).toEqual(5)
+      expect(foo.foobar(3)).toEqual(4)
+
+      val dyn = foo.asInstanceOf[js.Dynamic]
+      expect(dyn.bar(4, 5)).toEqual(9)
+      expect(dyn.bar(4)).toEqual(5)
+      expect(dyn.foobar(3)).toEqual(4)
+    }
```

Comment:
Could you add a test with dependent default params (`def foo(x: Int)(y: Int = x + 1)`) please?